### PR TITLE
Plans: decouple free-plan-paid-domain modal from the plans grid

### PR DESCRIPTION
**Pull request by aneeshd16**
Related to p1682942089709679-slack-CV2TX2PAN

## Proposed Changes

* This PR moves the `FreePlanPaidDomainDialog` component render to the `PlansFeaturesMain` component instead of the pricing grid. This would allow us to decouple the dialog from the grid since the dialog is not required in all contexts where the grid is used. 

## Testing Instructions

* Same as https://github.com/Automattic/wp-calypso/pull/75526

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/plan-features-2023-grid/components/features.tsx [hunk 1]
```diff
@@ -2,9 +2,9 @@ import { getPlanClass, FEATURE_CUSTOM_DOMAIN, isFreePlan } from '@automattic/cal
 import styled from '@emotion/styled';
 import classNames from 'classnames';
 import { useTranslate } from 'i18n-calypso';
+import { LoadingPlaceHolder } from '../../plans-features-main/components/loading-placeholder';
```
Comment (chriskmnds) [inline client/my-sites/plan-features-2023-grid/components/features.tsx:5]: This introduces a dependency on `plans-features-main/components/loading-placeholder`, which we'll need to refactor/extract (either move under `plan-features-2023-grid/components` or some other way).

We can ship and come back to. One more for the [list of TODOs](https://github.com/Automattic/wp-calypso/pull/76442) I guess 😬

File: client/my-sites/plans-features-main/components/free-plan-paid-domain-dialog.tsx [hunk 2]
```diff
@@ -8,9 +8,12 @@ import { useTranslate } from 'i18n-calypso';
 import { useState } from 'react';
 import { useSelector } from 'react-redux';
 import { getCurrentUserCurrencyCode } from 'calypso/state/currency-code/selectors';
-import usePlanPrices from '../plans/hooks/use-plan-prices';
-import { LoadingPlaceHolder } from './components/loading-placeholder';
-import { getQueryKey, useGetWordPressSubdomain } from './hooks/use-get-wordpress-subdomain';
+import {
+	getQueryKey,
+	useGetWordPressSubdomain,
+} from '../../plan-features-2023-grid/hooks/use-get-wordpress-subdomain';
```
Comment (chriskmnds) [inline client/my-sites/plans-features-main/components/free-plan-paid-domain-dialog.tsx:14]: These are the hooks extracted into the `DomainSuggestions` data store in the other PR 👍

**Review by chriskmnds — APPROVED** [review]
Thanks for making this refactor! It looks great and tests out. :-)

The only difference I notice from the media is the width of the description being a little narrower forcing it across two lines:

Screenshot 2023-05-04 at 2 16 17 PM

p.s. if we had rebased this PR onto https://github.com/Automattic/wp-calypso/pull/76454/files, then we'd be testing both only once 😆

**Comment by aneeshd16** [comment]
Thanks @chriskmnds for the review! I am investigating the failing E2E test.

**Comment by aneeshd16** [comment]
Hi @chriskmnds, can you please have a look at the new changes in https://github.com/Automattic/wp-calypso/pull/76532/commits/7d39823c5e2498bcd3ad247144420bbf406e3938?

Before this commit, the changes in the PR broke the domain upsell flow as the domain upsell flow depended on the redirection to happen here:
https://github.com/Automattic/wp-calypso/blob/4910db94dde21aced9111956d87aae3389df408d/client/my-sites/plan-features-2023-grid/index.tsx#L652

I also mentioned this in our TODO list (https://github.com/Automattic/wp-calypso/pull/76442#issuecomment-1532788484), so this is essentially a two for one! :D

**Comment by chriskmnds** [comment]
Testing. Weird I ran into this again (on the live image):

Screenshot 2023-05-04 at 3 49 33 PM

and also reaching `/start/plans` shows a label over the free plan:

Screenshot 2023-05-04 at 3 51 22 PM

I don't think anything related to here. some alternate reality

**Comment by chriskmnds** [comment]
Domain upsell flow works for me. per the instructions in Slack ( p1683204885105389-slack-CV2TX2PAN ), I can reach `/plans` with:

Screenshot 2023-05-04 at 4 08 33 PM

**Pull request merged**